plugins/module_utils/adguardhome/api.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2025 Markus Bergholz
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+import json
+
+import ansible.module_utils.six.moves.urllib.error as urllib_error  # pylint: disable=import-error
+from ansible.module_utils.urls import Request
+from ansible_collections.community.dns.plugins.module_utils.argspec import ArgumentSpec
+
+
+def create_adguardhome_argument_spec(required_if=None, additional_argument_specs=None):
+    always_argument_spec = {
+        'username': {'type': 'str', 'required': True},
+        'password': {'type': 'str', 'required': True, 'no_log': True},
+        'host': {'type': 'str', 'required': True},
+        'validate_certs': {'type': 'bool', 'default': True},
+    }
+    additional_argument_specs = {} if additional_argument_specs is None else additional_argument_specs
+    return ArgumentSpec(
+        required_if=required_if,
+        argument_spec={**always_argument_spec, **additional_argument_specs}
```

Comment:
Why not do something like
```suggestion
    if additional_argument_specs:
        always_argument_spec.update(additional_argument_specs)
    return ArgumentSpec(
        required_if=required_if,
        argument_spec=always_argument_spec,
```
? That works with all Python versions.